torch/distributed/pipelining/schedules.py

```diff
@@ -1241,21 +1231,28 @@ def _ready_to_schedule(
                 f"{rank=}, {len(compute_actions[rank])=}"
             )
             action = compute_actions[rank][0]
+            # handle new case where parent action (OVERLAP_F_B) can be comprised of subactions
+            if action is not None and action.sub_actions is not None:
+                all_actions = action.sub_actions
+            else:
+                all_actions = (action,)
 
-            if not _ready_to_schedule(action, prev_actions[rank]):
+            if not all(_ready_to_schedule(a, prev_actions[rank]) for a in all_actions):
                 continue
 
+            # The action's dependencies are satisfied, so add to schedule
             if action is not None:
                 comm_actions[rank].append(action)
-                prev_actions[rank].add(action)
-                if _has_comms(action):
-                    send, recv = _get_comms(action)
-                    # TODO we can avoid send/recv if the 2 stages are on the same rank.
-                    # should we avoid that in the runtime or here?
-                    comm_actions[rank].append(send)
-                    prev_actions[rank].add(send)
-                    comm_actions[stage_to_rank(recv.stage_index)].append(recv)
-                    prev_actions[stage_to_rank(recv.stage_index)].add(recv)
+                for a in all_actions:
```

Comment:
Isn't it taking us further from where we want to end up, to split the sub actions out? I would think you'd just plumb overlap_f_b all the way through to the runtime asap. It's easy to decompose it in the runtime into f,b actions if there isn't an actual overlap runner available. Then you can keep the ir passes simpler. Wdyt?

(I was mostly referring to appending the sub-action to the schedule, not the comms)